internal/engine/watchmanager_test.go

```diff
@@ -0,0 +1,148 @@
+package engine
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/windmilleng/tilt/internal/logger"
+	"github.com/windmilleng/tilt/internal/model"
+	"github.com/windmilleng/tilt/internal/ospath"
+	"github.com/windmilleng/tilt/internal/store"
+	"github.com/windmilleng/tilt/internal/testutils/tempdir"
+	"github.com/windmilleng/tilt/internal/watch"
+)
+
+func TestWatchesReappliedOnDockerIgnoreChange(t *testing.T) {
+	f := newWMFixture(t)
+	defer f.TearDown()
+
+	setManifestTarget := func(target model.DockerComposeTarget) {
+		m := model.Manifest{Name: "foo"}.WithDeployTarget(target)
+		mt := store.ManifestTarget{Manifest: m}
+		state := f.store.LockMutableStateForTesting()
+		state.UpsertManifestTarget(&mt)
+		state.WatchMounts = true
+		f.store.UnlockMutableState()
+		f.wm.OnChange(f.ctx, f.store)
+	}
+
+	target := model.DockerComposeTarget{Name: "foo"}
+	target.Mounts = []model.Mount{{LocalPath: f.Path()}}
+	setManifestTarget(target.WithIgnoredLocalDirectories([]string{f.JoinPath("bar")}))
```

Comment:
to be really thorough you'd probably want to test here that "bar" DOES get ignored?

Reply:
to be really thorough, that test should have been on the last PR and this file should also test all the other stuff in watchmanager :)